library/src/test/java/com/github/gfx/android/orma/test/QueryTest.java

```diff
@@ -364,6 +365,21 @@ public Book_Selector apply(Book_Selector it) {
     }
 
     @Test
+    public void whereComplexConditionGroup() throws Exception {
+        List<Book> books = db.selectFromBook()
+                .titleEq("friday")
+                .and()
+                .where(new Function1<Book_Selector, Book_Selector>() {
+                    @Override
+                    public Book_Selector apply(Book_Selector it) {
+                        return it.titleEq("today").or().priceGe(150);
+                    }
+                })
+                .toList();
+        assertThat(books, hasSize(0));
```

Comment:
Before this fix, this code generates `SELECT * FROM Book WHERE title = "friday" AND title = "today" OR price >= 150` and the result count is 1 ( "today" book ).

After the fix, this code generates `SELECT * FROM Book WHERE title = "friday" AND (title = "today" OR price >= 150)` and the result is empty.